cmd/kubeapps-apis/_test/mock_plugin.go

```diff
@@ -0,0 +1,273 @@
+Copyright © 2021 VMware
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package _test
+
+import (
+	"context"
+
+	corev1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	packages "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	plugins "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/plugins/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	GlobalPackagingCluster  = "default"
+	DefaultAppVersion       = "1.2.6"
+	DefaultCategory         = "cat-1"
+	DefaultIconURL          = "https://example.com/package.svg"
+	DefaultHomeURL          = "https://example.com/home"
+	DefaultRepoURL          = "https://example.com/repo"
+	DefaultDescription      = "description"
+	DefaultId               = "repo-1/package-id"
+	DefaultNamespace        = "my-namespace-1"
+	DefaultPkgVersion       = "1.0.0"
+	DefaultPkgUpdateVersion = "2.0.0"
+	DefaultReleaseName      = "my-release-1"
+	DefaultReleaseNamespace = "my-release-namespace-1"
+	DefaultReleaseNotes     = "some notes"
+	DefaultReleaseValues    = "{\"value\":\"new\"}"
+	DefaultReleaseVersion   = "1.2.3"
+	DefaultValuesSchema     = "\"$schema\": \"http://json-schema.org/schema#\""
+	DefaultReadme           = "#readme"
+	DefaultValues           = "key: value"
+	DefaultMaintainerName   = "me"
+	DefaultMaintainerEmail  = "me@example.com"
+)
+
+var defaultInstalledPackageStatus = &corev1.InstalledPackageStatus{
+	Ready:      true,
+	Reason:     corev1.InstalledPackageStatus_STATUS_REASON_INSTALLED,
+	UserReason: "ReconciliationSucceeded",
+}
+
+type TestPackagingPlugin struct {
+	packages.UnimplementedPackagesServiceServer
+	plugin                    *plugins.Plugin
+	availablePackageSummaries []*corev1.AvailablePackageSummary
+	availablePackageDetail    *corev1.AvailablePackageDetail
+	installedPackageSummaries []*corev1.InstalledPackageSummary
+	installedPackageDetail    *corev1.InstalledPackageDetail
+	packageAppVersions        []*corev1.PackageAppVersion
+	categories                []string
+	nextPageToken             string
+	status                    codes.Code
+}
+
+type MockPkgsPluginWithServer struct {
+	Plugin *plugins.Plugin
+	Server packages.PackagesServiceServer
+}
+
+func NewTestPackagingPlugin(plugin *plugins.Plugin) *TestPackagingPlugin {
+	return &TestPackagingPlugin{
+		plugin: plugin,
+	}
+}
+
+func MakeTestPackagingPlugin(
+	plugin *plugins.Plugin,
+	availablePackageSummaries []*corev1.AvailablePackageSummary,
```

Comment:
Up to you, but it might be easier to use if, rather than requiring all this data (and lack of data) when creating the test plugin, if the callsite can just create it without any data, then set what they want through public fields.